cartographer/sensor/internal/voxel_filter_test.cc

```diff
@@ -33,18 +34,37 @@ TEST(VoxelFilterTest, ReturnsOnePointInEachVoxel) {
                                 {{0.f, 0.f, 0.1f}}});
   const PointCloud result = VoxelFilter(point_cloud, 0.3f);
   ASSERT_EQ(result.size(), 2);
+  EXPECT_THAT(result.intensities(), IsEmpty());
   EXPECT_THAT(point_cloud.points(), Contains(result[0]));
   EXPECT_THAT(point_cloud.points(), Contains(result[1]));
   EXPECT_THAT(result.points(), Contains(point_cloud[2]));
 }
 
+TEST(VoxelFilterTest, CorrectIntensities) {
+  std::vector<RangefinderPoint> points;
+  std::vector<float> intensities;
+  for (int i = 0; i < 100; ++i) {
+    const float value = 0.1f * i;
+    points.push_back({{-100.f, 0.3f, value}});
+    intensities.push_back(value);
+  }
+  const PointCloud point_cloud(points, intensities);
+  const PointCloud result = VoxelFilter(point_cloud, 0.3f);
+
+  ASSERT_EQ(result.intensities().size(), result.points().size());
+  for (size_t i = 0; i < result.size(); ++i) {
+    ASSERT_NEAR(result[i].position.z(), result.intensities()[i], 1e-6);
```

Comment:
This confused me in the beginning.
So you applied a trick by setting the Z and intensity values for each point to an unique number for each entry of the point cloud, such that you can assert here that the pairs are still correct after the voxel filter - am I right?
Could be worth to mention this in a small comment.

Reply:
Added a comment. PTAL.

Reply:
Thank you